src/main/kotlin/com/openlattice/organizations/HazelcastOrganizationService.kt

```diff
@@ -75,7 +75,11 @@ class HazelcastOrganizationService(
 
     @Timed
     fun maybeGetOrganization(p: Principal): Optional<SecurablePrincipal> {
-        return securePrincipalsManager.maybeGetSecurablePrincipal(p)
+        return try {
```

Comment:
this feels very close to what `Optional.ofNullable()` does. Definitely prefer checking null over try-catching an exception

Reply:
I agree, although `spm.getPrincipal` is somewhat heavily used outside of this PR and I wasn't sure if any of those calls are specifically expecting an exception to be thrown in the null case. I can check/change those though